scripts/build/termux_download_deb_pac.sh

```diff
@@ -1,126 +1,79 @@
-					PKG_PATH=$(jq -r '."'$PACKAGE'"."FILENAME"' "${TERMUX_COMMON_CACHEDIR}-aarch64/$PACKAGE_FILE_PATH")
-					PKG_PATH="aarch64/${PKG_PATH}"
-				fi
-			fi
-			if [ -n "$PKG_HASH" ] && [ "$PKG_HASH" != "null" ]; then
-				if [ ! "$TERMUX_QUIET_BUILD" = true ]; then
-					if [ "$TERMUX_REPO_PKG_FORMAT" = "debian" ]; then
-						echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx-1]}/dists/${TERMUX_REPO_DISTRIBUTION[$idx-1]}"
-					elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-						echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx-1]}"
-					fi
-				fi
-				break
-			fi
-		elif [ -f "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/${PACKAGE_FILE_PATH}" ]; then
-			if [ "$TERMUX_REPO_PKG_FORMAT" = "debian" ]; then
-				read -d "\n" PKG_PATH PKG_HASH <<<$(./scripts/get_hash_from_file.py "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH" $PACKAGE $VERSION)
-			elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-				if [ "$TERMUX_WITHOUT_DEPVERSION_BINDING" = "true" ] || [ $(jq -r '."'$PACKAGE'"."VERSION"' "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH") = "${VERSION_PACMAN}" ]; then
-					PKG_HASH=$(jq -r '."'$PACKAGE'"."SHA256SUM"' "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH")
-					PKG_PATH=$(jq -r '."'$PACKAGE'"."FILENAME"' "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH")
-					PKG_PATH="${PACKAGE_ARCH}/${PKG_PATH}"
-				fi
-			fi
-			if [ -n "$PKG_HASH" ] && [ "$PKG_HASH" != "null" ]; then
-				if [ ! "$TERMUX_QUIET_BUILD" = true ]; then
-					if [ "$TERMUX_REPO_PKG_FORMAT" = "debian" ]; then
-						echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx-1]}/dists/${TERMUX_REPO_DISTRIBUTION[$idx-1]}"
-					elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-						echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx-1]}"
-					fi
-				fi
-				break
-			fi
+		elif [[ -f "$TERMUX_COMMON_CACHEDIR-$PACKAGE_ARCH/$PACKAGE_FILE_PATH" ]]; then
+			read -d "\n" PKG_PATH PKG_HASH <<< "$(termux_download_deb_pac_get_pkg_hash_and_version "$PACKAGE_ARCH")" || :
+			[[ -n "$PKG_HASH" && "$PKG_HASH" != "null" ]] && break || :
+		elif [[ -f "$TERMUX_COMMON_CACHEDIR-aarch64/$PACKAGE_FILE_PATH" ]]; then
+			# Packages file for $PACKAGE_ARCH do not exist. Could be an aptly mirror where the
+			# all arch is mixed into the other arches, check for package in aarch64 Packages instead.
+			read -d "\n" PKG_PATH PKG_HASH <<< "$(termux_download_deb_pac_get_pkg_hash_and_version aarch64)" || :
+			[[ -n "$PKG_HASH" && "$PKG_HASH" != "null" ]] && break || :
 		fi
 	done
 
-	if [ "$PKG_HASH" = "" ] || [ "$PKG_HASH" = "null" ]; then
+	if [[ "$PKG_HASH" = "" || "$PKG_HASH" = "null" ]]; then
 		return 1
 	fi
 
-	termux_download "${TERMUX_REPO_URL[${idx}-1]}/${PKG_PATH}" \
-				"${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/${PKG_FILE}" \
+	termux_download "${TERMUX_REPO_URL[${idx}]}/$PKG_PATH" \
+				"$TERMUX_COMMON_CACHEDIR-$PACKAGE_ARCH/$PKG_FILE" \
```

Comment:
I'm gonna need solid repeatable evidence that the brackets actually impact the runtime of this step to a significant degree.
This just makes things less readable for a gain of what, a couple microseconds?